### PR TITLE
Add a basic UI

### PR DESCRIPTION
**Pull request by diarisdiakite**
In this PR, I:
📃 Created an `app.rb` file that serves as the console app entry-point. It has methods that do the following:
✔️ List all books.
✔️ List all people.
✔️ Create a person (teacher or student, not a plain `Person`).
✔️ Create a book.
✔️ Create a rental.
✔️ List all rentals for a given person id.
📃 In the `main.rb` file I defined the entry point, this is a method called `main` that is invoked at the end of the file. it does the following:
✔️ Present the user with a list of options to perform.
✔️ Lets users choose an option.
✔️ If needed, ask for parameters for the option.
✔️ Have a way to quit the app.

📃 I Made sure that my console app behaves in the same way as it is in the example

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  def display_menu
+    puts 'Please select an option by entering a number'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a Book'
+    puts '5. Create a Rental'
+    puts '6. List all rentals for a given person id'
+    puts '7. Exit'
+  end
```
Comment (Shedrack-Sunday) [inline app.rb]: - [x] This part should be in a file named `main.rb`. It should be an entry point of your app, where you call the `app.rb` file to start your application. Kindly make adjustments based on the requirements below 👍 

### Requirements:
image

### Sample code snippet:
```
# main.rb
def main
  app = App.new()

  response = nil


  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
       call function.....
    when "2"
       call function.....
        .
        .
        .
   
    end

    puts "\n"
  end
end

main()
```

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      end
+    end
+  end
+end
+
+# rubocop:enable Metrics
+def main
+  app = App.new
+  app.run
+end
```
Comment (Shedrack-Sunday) [inline app.rb]: - [ ] As I stated above, this should go into your main.rb, to start the application. kindly make the changes 👍

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        print 'Age: '
+        age = gets.chomp.to_i
+        print 'Name: '
+        name = gets.chomp
+        if input == 1
+          print 'Classroom: '
+          classroom = gets.chomp
+          Student.add_a_student(age, name, classroom)
+          puts "The student #{name} has been created and added to the classroom #{classroom}"
+        elsif input == 2
```
Comment (Shedrack-Sunday) [inline app.rb]: - [ ] From the video guide, when you chose a student on option `1`, there's a parent permission option which yours is missing, instead a classroom option was asked. Kindly fix that 👍 

### Vidoe prompt:
image

### Your prompt:
image

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        puts "We have #{books_count} book."
+        if books_count.positive?
+          puts 'Here is the list of all books'
+          Book.list_all_books
+        end
+      when 2
+        puts "[Students] (#{students_count})"
+        Student.list_all_students
+        puts "[Teachers] (#{teachers_count})"
+        Teacher.list_all_teachers
```
Comment (Shedrack-Sunday) [inline app.rb]: - [ ] Kindly note that in displaying the list of persons, th key parameters are: `class(teacher or student)`, `name`, `id` and `age`. Yours has only `id` and `name`. Kindly fix the rmaining parameters:

### Video prmpt:
image

###Your prompt:
image

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            puts 'Select a person by its number from the following list: '
+            Person.list_all_people
+
+            selected_id = gets.chomp.to_i
+            person = Person.select_a_person(selected_id)
+
+            if person
+              date = Date.today
+              Rental.add_a_rental(date, book, person)
+              puts 'Rental created successfully'
```
Comment (Shedrack-Sunday) [inline app.rb]: - [ ] When adding a rental, there is a field to manually add a custom date. Yours is not present. If this is a Class issue from your rental class, please do fix it in order to have an input filed 👍 

### Video prompt:
image

### Your prompt:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @diarisdiakite          👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Ruby features ✔️
- Clean use of class calls ✔️  

# Required Changes  🌵  

- [x] The changes required are minor but very important so we avoid bugs in the application.

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app.rb (hunk 1 above)
Comment (diarisdiakite) [reply app.rb]: Done @Shedrack-Sunday, Thanks.

File: app.rb (hunk 2 above)
Comment (diarisdiakite) [reply app.rb]: Done @Shedrack-Sunday, Thanks.

File: app.rb (hunk 3 above)
Comment (diarisdiakite) [reply app.rb]: Done @Shedrack-Sunday, Thanks.

File: app.rb (hunk 4 above)
Comment (diarisdiakite) [reply app.rb]: Done @Shedrack-Sunday, Thanks.

File: app.rb (hunk 5 above)
Comment (diarisdiakite) [reply app.rb]: Done @Done @Shedrack-Sunday, Thanks.

File: main.rb [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+          puts 'Select a person by its number from the following list: '
+          Person.list_all_people
+
+          selected_index = gets.chomp.to_i
+          person = Person.select_a_person(selected_index)
+          puts ''
+
+          if person
+            date = Date.today
```
Comment (skyv26) [inline main.rb]: - [x] Well done by implementing the changes asked by your previous code reviewer except the last one, according to which, the date shouldn't be automatic, it should be manually added by the user. Please refactor your code in order to fulfil the requirement as shown in the example video provided by **Microverse**. Implementing the requirement will help you to become detail oriented developer.

File: main.rb [hunk 7]
```diff
@@ -0,0 +1,148 @@
+require 'date'
+require_relative 'library/classes/book'
+require_relative 'library/classes/person'
+require_relative 'library/classes/teacher'
+require_relative 'library/classes/student'
+require_relative 'library/classes/rental'
+
+# This class represents the main application for managing books, people, and rentals.
+
+# rubocop:disable Metrics
```
Comment (skyv26) [inline main.rb]: - [ ] You did a great job shifting your code from `app.rb` to `main.rb`. I know it wasn't easy to move your code and make the adjustments to meet the requirements and make your application run successfully like it was before. But as per the requirement, you should not disable `rubocop` in the project's code snippet. Please remove the comment and resolve the linter issue before submitting for the next review.

File: main.rb [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    when 7
+      break
+    else
+      puts 'Invalid choice'
+      puts ''
+    end
+  end
+end
+
+# rubocop:enable Metrics
```
Comment (skyv26) [inline main.rb]: - [ ] You did a great job shifting your code from `app.rb` to `main.rb`. I know it wasn't easy to move your code and make the adjustments to meet the requirements and make your application run successfully like it was before. But as per the requirement, you should not disable `rubocop` in the project's code snippet. Please remove the comment and resolve the linter issue before submitting for the next review.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @diarisdiakite, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implemented the features as asked by the previous code reviewer ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 6 above)
Comment (diarisdiakite) [reply main.rb]: Done @skyv26. Thanks.

File: main.rb (hunk 7 above)
Comment (diarisdiakite) [reply main.rb]: Done @skyv26. Rucocop has been re-enabled for the Metrics.

File: main.rb (hunk 8 above)
Comment (diarisdiakite) [reply main.rb]: Done @skyv26. Rucocop has been re-enabled for the Metrics.

**Review by topeogunleye — APPROVED** [review]
### Hi @diarisdiakite, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 

Hello!

Wow, you've been on a roll! 🚀 Let's dive into the positive updates on your project.

1. **Responsive to Feedback**: 
    - Your ability to quickly and accurately address feedback is impressive. This showcases not only your technical prowess but also your adaptability and commitment to quality. 👌

2. **Code Transition**: 
    - Shifting your code from `app.rb` to `main.rb` and ensuring everything still runs smoothly was no small feat. Nicely done! This change makes your code structure more in line with the best practices and recommendations. 📂

3. **Linting and Code Quality**:
    - Kudos for not bypassing the RuboCop guidelines and ensuring that your code adheres to the highest standards. This will go a long way in your journey as a developer. 🛠️🔍

4. **Date Input Feature**: 
    - Excellent work adjusting the date feature to be in line with the requirements. Now, with the manual input by the user, your application aligns perfectly with the expected behavior. 📅🖊️

#### Wrapping Up:

Your dedication to perfection and constant improvement is evident. It's genuinely inspiring to see how you've meticulously worked through each piece of feedback and fine-tuned your project to this level of excellence. 💎

Keep harnessing this spirit and approach in all your future projects. Onward and upward! 🚀

Remember, every line of code you refine and every feedback you integrate is a step closer to mastery. Keep it up! 👏👏👏

Warm regards and happy coding! 💼🖥️🎈
 
## [Optional] suggestions:
👉 One optional suggestion for improvement is to follow the convention for naming your branch. Usually, branch names should use the past tense verb to indicate that the feature has been implemented. For example, instead of `add_basic_ui`, you could use `added_basic_ui`. This would make your branch name more consistent and clear for future pull requests. 

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**